dotnet/src/Connectors/Connectors.Gemini/TextGeneration/GeminiTextGenerationService.cs

```diff
@@ -0,0 +1,121 @@
+using Microsoft.SemanticKernel.Connectors.Gemini.Settings;
+using Microsoft.SemanticKernel.Http;
+using Microsoft.SemanticKernel.Services;
+using Microsoft.SemanticKernel.TextGeneration;
+
+namespace Microsoft.SemanticKernel.Connectors.Gemini;
+
+/// <summary>
+/// Represents a service for generating text using the Gemini API.
+/// </summary>
+public sealed class GeminiTextGenerationService : ITextGenerationService
+{
+    private const string GeminiApiEndpointBase = "https://generativelanguage.googleapis.com/v1beta/models";
+
+    private readonly Dictionary<string, object?> _attributes = new();
+    private readonly string _model;
+    private readonly HttpClient _httpClient;
+    private readonly string? _apiKey;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="GeminiTextGenerationService"/> class.
+    /// </summary>
+    /// <param name="model">The model identifier.</param>
+    /// <param name="apiKey">The API key.</param>
+    /// <param name="httpClient">The optional HTTP client.</param>
+    public GeminiTextGenerationService(string model, string apiKey, HttpClient? httpClient = null)
+    {
+        Verify.NotNullOrWhiteSpace(model);
+        Verify.NotNullOrWhiteSpace(apiKey);
+
+        this._model = model;
+        this._apiKey = apiKey;
+        this._httpClient = HttpClientProvider.GetHttpClient(httpClient);
+        this._attributes.Add(AIServiceExtensions.ModelIdKey, this._model);
+    }
+
+    /// <inheritdoc />
+    public IReadOnlyDictionary<string, object?> Attributes => this._attributes;
+
+    /// <inheritdoc />
+    public async Task<IReadOnlyList<TextContent>> GetTextContentsAsync(
+        string prompt,
+        PromptExecutionSettings? executionSettings = null,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default)
+    {
+        using var httpRequestMessage = this.VerifyArgumentsAndGetHTTPRequestMessage(prompt, executionSettings);
+
+        using var response = await this._httpClient.SendWithSuccessCheckAsync(httpRequestMessage, cancellationToken).ConfigureAwait(false);
+        var body = await response.Content.ReadAsStringWithExceptionMappingAsync().ConfigureAwait(false);
+
+        var textGenerationResponse = JsonSerializer.Deserialize<TextGenerationResponse>(body);
+        if (textGenerationResponse is null)
+        {
+            throw new KernelException("Unexpected response from model")
+            {
+                Data = { { "ResponseData", body } },
+            };
+        }
+
+        return textGenerationResponse.Candidates.Select(c => new TextContent(c.Content.Parts[0].Text,
+            this.GetModelId(), textGenerationResponse)).ToList().AsReadOnly();
+    }
+
+    private HttpRequestMessage VerifyArgumentsAndGetHTTPRequestMessage(string prompt, PromptExecutionSettings? executionSettings)
+    {
+        Verify.NotNullOrWhiteSpace(prompt);
+        if (executionSettings is not GeminiPromptExecutionSettings geminiExecutionSettings)
```

Comment:
This should work with another type, including just the base. See what the OpenAI connector does. It tries to pull relevant info from the extra props dictionary, and everything else ends up just being a default.